src/praisonai/praisonai/ui/_auth.py

```
@@ -0,0 +1,169 @@
+        """Validate that UI credentials are safe for the bind host.
+        
+        Args:
+            bind_host: The host the UI server will bind to
+            username: Configured username
+            password: Configured password
+            allow_defaults: Whether to allow default credentials (escape hatch)
+            
+        Raises:
+            UIStartupError: If credentials are unsafe for external binding
+        """
+        is_default_creds = (username == "admin" and password == "admin")
```

Ruff (0.15.10)

[error] Possible hardcoded password assigned to: "password"

(S105)